server/test/integration/com/thoughtworks/go/server/dao/VersionInfoSqlMapDaoIntegrationTest.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2015 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.server.dao;
+
+import com.thoughtworks.go.domain.GOVersion;
+import com.thoughtworks.go.domain.VersionInfo;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.util.Date;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(locations = {
+        "classpath:WEB-INF/applicationContext-global.xml",
+        "classpath:WEB-INF/applicationContext-dataLocalAccess.xml",
+        "classpath:WEB-INF/applicationContext-acegi-security.xml"
+})
+public class VersionInfoSqlMapDaoIntegrationTest {
+    @Autowired
+    private VersionInfoSqlMapDao versionInfoSqlMapDao;
+
+    @Autowired
+    private DatabaseAccessHelper dbHelper;
+
+    @Before
+    public void setup() throws Exception {
+        dbHelper.onSetUp();
+        versionInfoSqlMapDao.deleteAll();
```

Comment:
Looks like dbHelper.onSetUp() and dbHelper.onTearDown() does not roll back, the tests fail upon removing the explicit deleteAll